### PR TITLE
Sort applicants

### LINKED ISSUE #24: Sort applicants by descriptor as the sorting criteria
As a HR Manager at an F&B outlet, I can sort the employee list by descriptor so that I can find relevant information quickly.

### PR DESCRIPTION
**Pull request by ivanleekk**
Implement `sort` command by `NAME` and by `PHONE`.
Closes #24 

Will require an update to change class names when #64 is merged

### CODE REVIEW

File: src/main/java/seedu/staffsnap/logic/commands/SortCommand.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        this.descriptor = descriptor;
+    }
+
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        model.updateSortedApplicantList(descriptor);
+        model.updateFilteredApplicantList(PREDICATE_HIDE_ALL_APPLICANTS);
+        model.updateFilteredApplicantList(PREDICATE_SHOW_ALL_APPLICANTS);
```
Comment (craigtonlian) [inline src/main/java/seedu/staffsnap/logic/commands/SortCommand.java:40]: This looks like a clever way update the list 🚀 

Not sure if there'll be any side-effects to this down the road, perhaps we can add in a comment to explain the purpose of this workaround for now. 

Down the line when we implement the filter() function, we might encounter a similar issue and have to revisit this.

File: src/main/java/seedu/staffsnap/model/applicant/Descriptor.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package seedu.staffsnap.model.applicant;
+
+/**
+ * Enum for valid Descriptors
+ */
+public enum Descriptor {
```
Comment (craigtonlian) [inline src/main/java/seedu/staffsnap/model/applicant/Descriptor.java:6]: I like the choice to set this up as an enum, since it will be immutable 😄

**Review by craigtonlian — CHANGES_REQUESTED** [review]
Other than the comment to add for `SortCommand#execute()`, it looks good! Thanks, Ivan! 😀

File: src/main/java/seedu/staffsnap/model/applicant/Descriptor.java (hunk 2 above)
Comment (ivanleekk) [reply src/main/java/seedu/staffsnap/model/applicant/Descriptor.java:6]: Thank you Craig!

File: src/main/java/seedu/staffsnap/logic/commands/SortCommand.java (hunk 1 above)
Comment (ivanleekk) [reply src/main/java/seedu/staffsnap/logic/commands/SortCommand.java:48]: Nice catch Craig, forgot that this was unintuitive.

Updated it with a comment describing why it does this, and potential fix in the future when filter has to be implemented.

**Review by craigtonlian — APPROVED** [review]
Lgtm 🥲

**Pull request merged**